File: utils-common/src/com/xenoage/utils/document/command/StopperCommand.java
package com.xenoage.utils.document.command;

/**
 * Base class for a {@link Command}, which is a stopper in the command history
 * (see {@link Undoability#Stopper}).
 * 
 * @author Andreas Wenger
 */
public abstract class StopperCommand
	implements Command {

	@Override public Undoability getUndoability() {
		return Undoability.Stopper;
	}

	@Override public void undo() {
		throw new IllegalStateException("not undoable");
	}

}

File: utils-common/src/com/xenoage/utils/document/command/StopperCommand.kt
package com.xenoage.utils.document.command

/**
 * Base class for a [Command], which is a stopper in the command history
 * (see [Undoability.Stopper]).
 */
abstract class StopperCommand(override val name: String) : Command {

	override val undoability: Undoability
		get() = Undoability.Stopper

	override fun undo() {
		throw IllegalStateException("not undoable")
	}

}
